contrib/pg_tde/src/access/pg_tde_tdemap.c

```diff
@@ -1006,19 +1006,13 @@ pg_tde_get_key_from_file(const RelFileLocator *rlocator, uint32 key_type, bool n
 	principal_key = GetPrincipalKey(rlocator->dbOid, LW_SHARED);
 	if (principal_key == NULL)
 	{
-		LWLockRelease(lock_pk);
-		if (no_map_ok)
-		{
-			return NULL;
-		}
-		ereport(ERROR,
-				(errmsg("failed to retrieve principal key. Create one using pg_tde_set_principal_key before using encrypted tables.")));
+		return NULL;
```

Comment:
Should be LWLockRelease(lock_pk);

Reply:
Ouch, good that you spotted this. This happened due to the code being broken out from a bigger commit.